### PR TITLE
chore: cleanup in vortex-ipc & vortex-file

### PR DESCRIPTION
**Pull request by gatesn**

### CODE REVIEW

File: vortex-file/src/read/mod.rs [hunk 1]
```diff
@@ -33,6 +34,7 @@ use vortex_ipc::stream_writer::ByteRange;
 pub use crate::read::mask::RowMask;
 
 // Recommended read-size according to the AWS performance guide
+// FIXME(ngates): this is dumb
```
Comment (robert3005) [inline vortex-file/src/read/mod.rs:37]: Why?

File: vortex-file/src/read/reader.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+use crate::{InitialRead, LayoutMessageCache, VortexFileArrayStream};
+
+pub struct VortexFileArrayReader<R> {
+    read: R,
+    initial: InitialRead,
+
+    message_cache: Arc<RwLock<LayoutMessageCache>>,
+}
+
+impl<R: VortexReadAt> VortexFileArrayReader<R> {
```
Comment (robert3005) [inline vortex-file/src/read/reader.rs:19]: Why is this here if it's todo?

File: vortex-file/src/read/mod.rs (hunk 1 above)
Comment (gatesn) [reply vortex-file/src/read/mod.rs:37]: It's a hard-coded constant, we don't even know if we're reading from S3 or not

File: vortex-file/src/read/stream.rs [hunk 3]
```diff
@@ -84,6 +88,7 @@ impl<R: VortexReadAt + Unpin> VortexFileArrayStream<R> {
     }
 
     pub fn dtype(&self) -> &DType {
+        // FIXME(ngates): why is this allowed to unwrap?
```
Comment (robert3005) [inline vortex-file/src/read/stream.rs:91]: This is allowed to unwrap because the caller will know whether it will throw. I have a change that wil get rid of the error here

**Review by robert3005 — APPROVED** [review]
